src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -31,39 +53,36 @@ public MoveFaultsToErrorQueueBehavior(CriticalError criticalError, IPipelineBase
             {
                 throw;
             }
-            catch (Exception exception)
+            catch (Exception e)
             {
-                try
-                {
-                    var message = context.Message;
+                faultsStatusStorage.AddException(uniqueMessageId, e);
```

Comment:
We are not re-throwing the caught exception. The only thing that we do it is copy certain information to message headers. So I think that we will not benefit from using ExceptionDispatchInfo.